src/Microsoft.AspNetCore.Server.IISIntegration.Tools/PublishIISCommand.cs

```diff
@@ -67,9 +73,14 @@ private string GetApplicationBasePath()
             return Directory.GetCurrentDirectory();
         }
 
-        private string GetApplicationName(string applicationBasePath)
+        private static ProjectContext GetProjectContext(string applicationBasePath, string framework)
         {
-            return ProjectReader.GetProject(Path.Combine(applicationBasePath, "project.json")).Name;
+            var project = ProjectReader.GetProject(Path.Combine(applicationBasePath, "project.json"));
+
+            return new ProjectContextBuilder()
+                .WithProject(project)
+                .WithTargetFramework(framework)
+                .Build();
         }
 
         private static bool ConfigureForAzure()
```

Comment:
It could be a parameter. In practice it won't be used since in majority of cases this tool is run as postpublishscript so you would have to modify the project.json file each time you want to publish when you know if the app will be hosted on Azure or not. Currently VS just sets the env variable when publishing without having to modify project.json. 
If you feel strongly about this feel free to file a bug - it's a different issue (and most likely not RC2)